spring-amqp/src/main/java/org/springframework/amqp/core/Message.java

```diff
@@ -107,11 +120,15 @@ private String getBodyContentAsString() {
 				return SerializationUtils.deserialize(new ByteArrayInputStream(this.body), whiteListPatterns,
 						ClassUtils.getDefaultClassLoader()).toString();
 			}
+			String encoding = this.messageProperties.getContentEncoding();
```

Comment:
As I see, messageProperties could be null here.

Reply:
Probably not in practice, but good catch; thanks.